ori/volume/server/volume.go

```diff
@@ -16,235 +16,150 @@ package server
+		return nil, fmt.Errorf("error getting ori metadata: %w", err)
 	}
 
-	imageKey, err := s.getImageKeyFromPV(pv)
+	spec, err := s.getOriVolumeSpec(image)
 	if err != nil {
-		return nil, fmt.Errorf("failed to provide image name: %w", err)
+		return nil, fmt.Errorf("error getting ori resources: %w", err)
 	}
 
-	wwn, err := generateWWN(s.wwnPrefix)
+	state, err := s.getOriState(image.Status.State)
 	if err != nil {
-		return nil, fmt.Errorf("unable to generate wwn: %w", err)
+		return nil, fmt.Errorf("error getting ori state: %w", err)
 	}
 
-	credentials, err := s.getVolumeAccessCredentials(ctx)
-	if err != nil {
-		return nil, fmt.Errorf("unable to get volume access credentials: %w", err)
+	var access *ori.VolumeAccess
+	if state == ori.VolumeState_VOLUME_AVAILABLE {
+		access, err = s.getOriVolumeAccess(ctx, log, image)
+		if err != nil {
+			return nil, fmt.Errorf("error getting ori volume access: %w", err)
+		}
 	}
 
-	return &ori.VolumeAccess{
-		Driver: s.driver,
-		Handle: wwn,
-		Attributes: map[string]string{
-			"monitors": strings.Join(monitors, ","),
-			"image":    imageKey,
-		},
-		SecretData: map[string][]byte{
-			userIDKey:  []byte(s.rookPoolName),
-			userKeyKey: credentials,
+	return &ori.Volume{
+		Metadata: metadata,
+		Spec:     spec,
+		Status: &ori.VolumeStatus{
+			State:  state,
+			Access: access,
 		},
 	}, nil
 }
 
-func (s *Server) clientGetVolumeClassFunc(ctx context.Context) func(string) (*storagev1alpha1.VolumeClass, error) {
-
-	return func(name string) (*storagev1alpha1.VolumeClass, error) {
-		volumeClass := &storagev1alpha1.VolumeClass{}
-		if err := s.client.Get(ctx, client.ObjectKey{Namespace: s.namespace, Name: name}, volumeClass); err != nil {
-			return nil, err
-		}
-		return volumeClass, nil
-	}
-}
-
-func (s *Server) getMonitorList(ctx context.Context) ([]string, error) {
-	rookConfigMap := &corev1.ConfigMap{}
-	if err := s.client.Get(ctx, types.NamespacedName{Name: s.rookPoolMonitorConfigmap, Namespace: s.rookNamespace}, rookConfigMap); err != nil {
-		return nil, fmt.Errorf("failed to get ceph monitors configMap %s: %w", client.ObjectKeyFromObject(rookConfigMap), err)
-	}
-	var list ceph.ClusterList
-	if val, ok := rookConfigMap.Data[monitorConfigMapDataKey]; !ok {
-		return nil, fmt.Errorf("unable to find data key %s in rook configMap %s", monitorConfigMapDataKey, client.ObjectKeyFromObject(rookConfigMap))
-	} else if err := json.Unmarshal([]byte(val), &list); err != nil {
-		return nil, fmt.Errorf("failed to decode ceph cluster list in rook config map %s: %w", client.ObjectKeyFromObject(rookConfigMap), err)
-	}
-	var monitors []string
-	for _, cluster := range list {
-		if cluster.ClusterID == s.rookClusterName {
-			monitors = cluster.Monitors
-			break
-		}
-	}
-	if len(monitors) == 0 {
-		return nil, fmt.Errorf("no monitors provided for clusterID %s", s.rookPoolName)
-	}
-	return monitors, nil
+func (s *Server) generateWWN() string {
+	wwn := strings.Replace(s.idGen.Generate(), "-", "", -1)
```

Comment:
Two things:
1. `IDGen` by default doesn't return `-` in its output, so the `strings.Replace` is not necessary.
2. I'd recommend using a separate instance of `IDGen` for WWN generation. This instance should then by default produce strings of length 16 which also makes the `[:16]` unnecessary